### PR TITLE
Feat/error handler

### PR DESCRIPTION
**Pull request by iamnovichek**
1. Error logger bot added
2. Error logging for zkLend and Nostra added

### CODE REVIEW

File: data_handler/error_handler/notifications.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        elif self._get_unique_message(message):
+            await self.bot.send_message(chat_id=ERROR_CHAT_ID, text=message)
+            self.SESSION_MESSAGES[self.SESSION_ID].append(Message(text=message, is_sent=True))
+            await self.bot.close()
+
+        else:
+            self.SESSION_MESSAGES[self.SESSION_ID].append(Message(text=message, is_sent=False))
+
+
+my_bot = TelegramBot(TELEGRAM_TOKEN)
```
Comment (djeck1432) [inline data_handler/error_handler/notifications.py:56]: `my_bot` rename it and move this init to `__init__.py`

File: data_handler/error_handler/values.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from dataclasses import dataclass
+from enum import Enum
+
+
+@dataclass
+class RedisSetKeys:
```
Comment (djeck1432) [inline data_handler/error_handler/values.py:6]: Do we still use it?

File: data_handler/handlers/helpers.py [hunk 3]
```diff
@@ -228,7 +232,14 @@ def load_data(
 
 
 # TODO: Improve this.
-def get_symbol(address: str) -> str:
+def get_symbol(address: str, protocol: str | None = None) -> str:
+    """
+    Get the symbol of the token by its address.
```
Comment (djeck1432) [inline data_handler/handlers/helpers.py:237]: update docstring, with adding info about sending error message to telegram bot

File: data_handler/handlers/helpers.py [hunk 4]
```diff
@@ -238,6 +249,16 @@ def get_symbol(address: str) -> str:
     for symbol, addr in symbol_address_map.items():
         if int(addr, base=16) == n:
             return symbol
+
+    # A tuple of that always has this order: `address`, `protocol`.
+    error_info = (address, protocol)
```
Comment (djeck1432) [inline data_handler/handlers/helpers.py:254]: why you can put it in set in the beginning?

File: data_handler/handlers/helpers.py [hunk 5]
```diff
@@ -238,6 +249,16 @@ def get_symbol(address: str) -> str:
...
+
+    if protocol and error_info not in ERROR_LOGS:
+        ERROR_LOGS.update({error_info})
+        asyncio.run(my_bot.send_message(
```
Comment (djeck1432) [inline data_handler/handlers/helpers.py:258]: don't forget to renamte `my_bot`

File: data_handler/handlers/helpers.py (hunk 4 above)
Comment (iamnovichek) [reply data_handler/handlers/helpers.py:254]: should i move it into the `if` statement ?

File: data_handler/.env.example [hunk 6]
```diff
@@ -8,3 +8,6 @@ DB_PASSWORD=#
 DB_HOST=db
 DB_PORT=#
 REDIS_HOST=redis
+
+# Telegram Error logger
```
[the remark this answers is not in the document]
Comment (iamnovichek) [reply data_handler/.env.example:12]: Done

File: data_handler/error_handler/notifications.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+from .values import Message
+from .config import TELEGRAM_TOKEN, ERROR_CHAT_ID
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+dispatcher = Dispatcher()
+
+
+class TelegramBot:
```
[the remark this answers is not in the document]
Comment (iamnovichek) [reply data_handler/error_handler/notifications.py:15]: Done

File: data_handler/handlers/helpers.py (hunk 5 above)
Comment (iamnovichek) [reply data_handler/handlers/helpers.py:258]: Done

File: data_handler/handlers/helpers.py (hunk 4 above)
Comment (iamnovichek) [reply data_handler/handlers/helpers.py:254]: Done

File: data_handler/error_handler/notifications.py (hunk 1 above)
Comment (iamnovichek) [reply data_handler/error_handler/notifications.py:56]: Done

File: data_handler/error_handler/notifications.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        messages = self.SESSION_MESSAGES[self.SESSION_ID]
+        not_sent_messages = list(filter(lambda message: not message.is_sent, messages))
+
+        for message in not_sent_messages:
+            if message.text != new_message:
+                return new_message
+
+    async def send_message(self, message: str) -> None:
+        """
+        Send a message to a chat.
```
[the remark this answers is not in the document]
Comment (iamnovichek) [reply data_handler/error_handler/notifications.py:37]: Done

File: data_handler/handlers/helpers.py (hunk 3 above)
Comment (iamnovichek) [reply data_handler/handlers/helpers.py:237]: Done

**Review by djeck1432 — APPROVED** [review]

**Pull request merged**